index.js

```diff
@@ -105,6 +105,24 @@
     };
   }
 
+  // type Iteration a = { value :: a, done :: Boolean }
+
+  //  iteratoinNext :: a -> Iteration a
+  function iteratoinNext(x) {
+    return {
+      value: x,
+      done: false
+    };
+  }
```

Comment:
We have a typo: `:%s/iteratoinNext/iterationNext/g`. Also, I'd prefer more compact formatting:

```javascript
  function iterationNext(x) { return {value: x, done: false}; }
```